lib/ssh/src/ssh_agent.erl

```diff
@@ -0,0 +1,134 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2019. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+%% Reference: https://tools.ietf.org/html/draft-miller-ssh-agent-02
+
+-module(ssh_agent).
+
+-behaviour(ssh_client_key_api).
+
+-include("ssh.hrl").
+-include("ssh_agent.hrl").
+
+-export([send/2]).
+-export([add_host_key/3, is_host_key/4, user_key/2]).
+
+-export_type([socket_path_option/0, timeout_option/0]).
+
+-type socket_path_option() :: {socket_path,  string()}.
+-type timeout_option() :: {timeout, integer()}.
+
+%% ssh_client_key_api implementation
+
+add_host_key(Host, Key, Opts) ->
+    ssh_file:add_host_key(Host, Key, Opts).
+
+is_host_key(Key, PeerName, Algorithm, Opts) ->
+    ssh_file:is_host_key(Key, PeerName, Algorithm, Opts).
```

Comment:
Just a thought: if someone makes a special ssh_file by some reason, they probably want that to be called from ssh_agent instead.
So an option {call_ssh_file, SshFile} could be useful -- with a better name maybe:

```suggestion
add_host_key(Host, Key, Opts) ->
    KeyCbOpts = proplists:get_value(key_cb_private, Opts, []),
    SshFileCb = proplists:get_value(call_ssh_file, KeyCbOpts, ssh_file),
    SshFileCb:add_host_key(Host, Key, Opts).

is_host_key(Key, PeerName, Algorithm, Opts) ->
    KeyCbOpts = proplists:get_value(key_cb_private, Opts, []),
    SshFileCb = proplists:get_value(call_ssh_file, KeyCbOpts, ssh_file),
    SshFileCb:is_host_key(Key, PeerName, Algorithm, Opts).
```